antlr_plsql/ast.py

```diff
@@ -205,148 +364,100 @@ def aggregateResult(self, aggregate, nextResult):
 
     def visitLimit_clause(self, ctx):
-        return  self.visitChildren(ctx, predicate = lambda n: n is not ctx.LIMIT())
+        return self.visitChildren(ctx, predicate=lambda n: n is not ctx.LIMIT())
 
     def visitColumn_alias(self, ctx):
-        return self.visitChildren(ctx, predicate = lambda n: n is not ctx.AS())
-
-    def visitTable_ref_list(self, ctx):
-        return self.visitChildren(ctx, predicate = lambda n: not isinstance(n, Tree.TerminalNodeImpl))
-
-    def visitGroup_by_clause(self, ctx):
-        return self.visitChildren(ctx, predicate = lambda n: not isinstance(n, Tree.TerminalNodeImpl))
+        return self.visitChildren(ctx, predicate=lambda n: n is not ctx.AS())
 
     def visitHaving_clause(self, ctx):
-        return  self.visitChildren(ctx, predicate = lambda n: n is not ctx.HAVING())
+        return self.visitChildren(ctx, predicate=lambda n: n is not ctx.HAVING())
+
+    # TODO similar pattern, reuse?
 
     def visitExpression_list(self, ctx):
         return [self.visit(expr) for expr in ctx.expression()]
-        #return self.visitChildren(ctx, predicate = lambda n: not isinstance(n, Tree.TerminalNode))
+        # return self.visitChildren(ctx, predicate = lambda n: not isinstance(n, Tree.TerminalNode))
 
     def visitInto_clause(self, ctx):
         return [self.visit(expr) for expr in ctx.variable_name()]
 
-    def visitJoin_on_part(self, ctx):
-        return self.visitChildren(ctx, predicate = lambda n: not isinstance(n, Tree.TerminalNode))
-
-    def visitJoin_using_part(self, ctx):
-        return self.visitChildren(ctx, predicate = lambda n: not isinstance(n, Tree.TerminalNode))
-
-    # removing parentheses ----------------------------------------------------
-
-    @staticmethod
-    def _exclude_parens(node):
-        return not isinstance(node, Tree.TerminalNodeImpl)
-
-    def visitAtom(self, ctx): 
-        return self.visitChildren(ctx, predicate = self._exclude_parens)
-
-    visitParenExpr = visitAtom
-    visitParenBinaryExpr = visitAtom
-
-# TODO: for some reason can't get this to work as in tsql
-#import inspect
-#for item in list(globals().values()):
-#    if inspect.isclass(item) and issubclass(item, AstNode):
-#        item._bind_to_visitor(AstVisitor)
+    _remove_terminal = [
+        "from_clause",
+        "table_ref_list",
+        "group_by_clause",
+        "join_on_part",
+        "join_using_part",
+        "atom",
+        "ParenExpr",
+        "ParenBinaryExpr",
+        "case_else_part",
+        "table_alias",
+        "subquery_factoring_clause",
+        "dml_table_expression_clause",
+        "function_argument",
+        "argument_list",
+    ]
+
+
+# TODO: convert more visitors now visitor generation is fixed
+# for some reason can't get this to work as in tsql
+# Override visit methods in AstVisitor for all nodes (in _rules) that convert to the AstNode classes
+import inspect
+
+for item in list(globals().values()):
```

Comment:
As discussed in PM there's several ways to clean this up. However It can happen in a different PR. Maybe Meta class for the `_bind_to_visitor`? Maybe there's even easier ways.

For the dynamic method assignment below, I'd consider just writing the functions out with a helper function.